src/tof/model.py

```diff
@@ -62,177 +78,187 @@ def run(self, npulses: int = 1):
         initial_mask = sc.ones(
             sizes=self.pulse.birth_times.sizes, unit=None, dtype=bool
         )
-        for comp in components:
+
+        result_choppers = {}
+        result_detectors = {}
+        for c in components:
+            comp = Component(**c.to_dict())
```

Comment:
Took me a while to figure out that you are converting detectors and choppers to components. Some comments on the overall mechanism here (in `run`) would be useful.